ctp2_code/ai/strategy/scheduler/scheduler.cpp

```diff
@@ -686,7 +683,7 @@ void Scheduler::Match_Resources(const bool move_armies)
 
 			// Actually should be checked in the next cycle, but there still seems to be something wrong.
 			Goal_List::iterator tmp_goal_iter = goal_iter;
-			--goal_iter;
+			goal_iter++;
 			m_goals.erase(tmp_goal_iter);
```

Comment:
That could be simplified, list::erase returns the iterator after the one to be erased. The code should then be a one liner, with eliminating the temporary iterator: goal_iter = m_goals.erase(goal_iter);

The commented lines after this can also go.